### DISCUSSION TITLE
Mempool.space explorer

### DISCUSSION THREAD

**Discussion by astupidmoose** [btcpayserver/btcpayserver#2821] [General]
I've been using the mempool.space explorer a lot more frequently compared to the current blockstream.info explorer linked by default in btcpay. 

At the very least as a first step, I think it might be nice to have a setting that chooses between these two explorers (If blockstream, link to the transaction using their link, but if mempool.space, link using their link). 

Ultimately, as it's an OSS project, the nodes could have the ability to run their own instance of mempool.space like is done on Umbrel, but possibly using custom URL's which give a bit more of an "enclosed ecosystem" where merchants can use the explorer with your own trusted node when sending information to customers or even internally.

**Comment by dennisreimann** [comment]
Great idea, I can take a look!

**Comment by Kukks** [comment]
> I've been using the mempool.space explorer a lot more frequently compared to the current blockstream.info explorer linked by default in btcpay.
> 
> At the very least as a first step, I think it might be nice to have a setting that chooses between these two explorers (If blockstream, link to the transaction using their link, but if mempool.space, link using their link).
> 
> Ultimately, as it's an OSS project, the nodes could have the ability to run their own instance of mempool.space like is done on Umbrel, but possibly using custom URL's which give a bit more of an "enclosed ecosystem" where merchants can use the explorer with your own trusted node when sending information to customers or even internally.

You can already do this: `Server Settings=>Policies=>Override the block explorers used`

**Reply by Kukks** [reply]
Also: https://github.com/btcpayserver/btcpayserver-docker/pull/516

**Reply by astupidmoose** [reply]
Awesome, didn't know this existed but have changed it now! 😃

**Comment by astupidmoose** [comment]
Following up on this original comment. 

Mempool seems to be gaining even more popularity. Would be nice to be able to have a mempool instance, and a domain setting for mempool that can run using the nodes internal explorer (I think it also requires electrum for address looking), but would be nice to use the BTCPay node, along with the mempool instance on the same server. 

For instance, we would setup mempool.coincards.com or something similar, and the link would be our own mempool instance on our node, using our btcpay servers chain files.

**Reply by dennisreimann** [reply]
We can try to revive @Kukks docker fragment pull request (btcpayserver/btcpayserver-docker#516). Meanwhile #4100 should also land soon.
